File: Python/01.ProgrammingBasics/06.Conditional-Statements-Adv-Exercise/04.FishingBoat/_04.FishingBoat.py
budget = int(input())
season = input()
fishermen_count = int(input())

boat_rental_prices = {
    "Spring": 3000,
    "Summer": 4200,
    "Autumn": 4200,
    "Winter": 2600
}

discounts = {
    "up_to_6": 0.10,
    "7_to_11": 0.15,
    "12_and_above": 0.25
}

boat_rental = boat_rental_prices[season]
discount = 0

if fishermen_count <= 6:
    discount = discounts["up_to_6"]
elif 7 <= fishermen_count <= 11:
    discount = discounts["7_to_11"]
else:
    discount = discounts["12_and_above"]

total_cost = boat_rental - (boat_rental * discount)

if fishermen_count % 2 == 0 and season != "Autumn":
    total_cost *= 0.95

if budget >= total_cost:
    left_money = budget - total_cost
    print(f"Yes! You have {left_money:.2f} leva left.")
else:
    needed_money = total_cost - budget
    print(f"Not enough money! You need {needed_money:.2f} leva.")

File: C++/01-Basics/03.3-ConditionalStatementsAdvЕxercise/04.FishingBoat/04.FishingBoat.cpp
#include <iostream>
#include <string>
#include <iomanip>

using namespace std;

int main() {
    int budget;
    string season;
    int fishermen;
    cin >> budget >> season >> fishermen;

    double shipRent = 0.0;

    if (season == "Spring") {
        shipRent = 3000.0;
    }
    else if (season == "Summer" || season == "Autumn") {
        shipRent = 4200.0;
    }
    else if (season == "Winter") {
        shipRent = 2600.0;
    }

    if (fishermen <= 6) {
        shipRent *= 0.90;
    }
    else if (fishermen <= 11) {
        shipRent *= 0.85;
    }
    else {
        shipRent *= 0.75;
    }

    if (fishermen % 2 == 0 && season != "Autumn") {
        shipRent *= 0.95;
    }

    double diff = abs(budget - shipRent);

    if (budget >= shipRent) {
        cout << fixed << setprecision(2) << "Yes! You have " << diff << " leva left." << endl;
    }
    else {
        cout << fixed << setprecision(2) << "Not enough money! You need " << diff << " leva." << endl;
    }

    return 0;
}
